go/teams/chain.go

```diff
@@ -845,6 +852,11 @@ func (t *TeamSigChainPlayer) addInnerLink(
 			return res, fmt.Errorf("signer is not an owner: %v (%v)", signer, team.Members.Owners)
 		}
 
+		if settings := team.Settings; settings != nil {
+			err = t.parseTeamSettings(settings, &res.newState)
+			return res, err
+		}
```

Comment:
I like this way of phrasing the control structure is a little convoluted. Especially because it early returns.

How about this?
```
if team.Settings != nil {
  err = t.parseTeamSettings(team.Settings, &res.newState)
  if err != nil {
    return res, err
  }
}
```